### PR TITLE
Create qgis.profile

### PR DESCRIPTION
**Pull request by jose1711**

### CODE REVIEW

File: etc/qgis.profile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Firejail profile for qgis
+# Description: GIS application
+# This file is overwritten after every install/update
+# Persistent local customizations
+include qgis.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.config/QtProject.conf
```
Comment (rusty-snake) [inline etc/qgis.profile]: Add here: `noblacklist ${HOME}/.local/share/QGIS`
Add to `disable-programs.inc`: `blacklist ${HOME}/.local/share/QGIS`

File: etc/qgis.profile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+whitelist ${HOME}/.local/share/QGIS/
```
Comment (rusty-snake) [inline etc/qgis.profile]: remove trailing `/` and add above `mkdir ${HOME}/.local/share/QGIS`

File: etc/qgis.profile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+nodbus
+nodvd
+nogroups
+nonewprivs
+noroot
+nosound
+notv
+nou2f
+novideo
+seccomp
```
Comment (rusty-snake) [inline etc/qgis.profile]: Add `protocol unix,inet,inet6`

Comment (jose1711) [reply etc/qgis.profile]: > Add `protocol unix,inet,inet6`

with this included in the profile QGIS was logging some errors so I decided to leave it out

File: etc/qgis.profile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+shell none
+tracelog
+
+disable-mnt
+private-cache
+private-dev
```
Comment (rusty-snake) [inline etc/qgis.profile]: Add `private-etc alternatives,ca-certificates,crypto-policies,fonts,machine-id,pki,resolv.conf,ssl,QGIS,QGIS.conf,Trolltech.conf`

File: etc/qgis.profile [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+whitelist ${HOME}/.local/share/QGIS/
+include whitelist-common.inc
+include whitelist-var-common.inc
+
+caps.drop all
+netfilter
```
Comment (rusty-snake) [inline etc/qgis.profile]: add `machine-id`

File: etc/qgis.profile (hunk 3 above)
Comment (rusty-snake) [inline etc/qgis.profile]: qgis don't start with `seccomp` on my system.

Comment (rusty-snake) [reply etc/qgis.profile]: @jose1711 can you paste?

**Comment by rusty-snake** [comment]
Great profile, but I'm not sure if it should be a whitelisting profile and have disable-mnt, because then you can't store thing on mobile devices or in your HOME.

File: etc/qgis.profile (hunk 3 above)
Comment (jose1711) [reply etc/qgis.profile]:
```Critical: Could not create AF_NETLINK socket (Operation not supported)
Critical: Could not create AF_NETLINK socket (Operation not supported)
Critical: Could not create AF_NETLINK socket (Operation not supported)
Critical: Could not create AF_NETLINK socket (Operation not supported)
Critical: Could not create AF_NETLINK socket (Operation not supported)
Exception:  403 Client Error: Forbidden for url: https://www.amigocloud.com/api/v1/me/projects?summary```

the last line is related to a 3rd party plugin

Comment (rusty-snake) [reply etc/qgis.profile]: OK, then add `netlink` to the protocol list.

Comment (jose1711) [reply etc/qgis.profile]: > qgis don't start with `seccomp` on my system.

interesting, it does work for me on a fresh arch linux system

Comment (rusty-snake) [reply etc/qgis.profile]: the default list without `mbind` works:
```
seccomp.drop _sysctl,acct,add_key,adjtimex,afs_syscall,bdflush,bpf,break,chroot,clock_adjtime,clock_settime,create_module,delete_module,fanotify_init,finit_module,ftime,get_kernel_syms,getpmsg,gtty,init_module,io_cancel,io_destroy,io_getevents,io_setup,io_submit,ioperm,iopl,ioprio_set,kcmp,kexec_file_load,kexec_load,keyctl,lock,lookup_dcookie,migrate_pages,modify_ldt,mount,move_pages,mpx,name_to_handle_at,nfsservctl,ni_syscall,open_by_handle_at,pciconfig_iobase,pciconfig_read,pciconfig_write,perf_event_open,personality,pivot_root,process_vm_readv,process_vm_writev,prof,profil,ptrace,putpmsg,query_module,reboot,remap_file_pages,request_key,rtas,s390_mmio_read,s390_mmio_write,s390_runtime_instr,security,set_mempolicy,setdomainname,sethostname,settimeofday,sgetmask,ssetmask,stime,stty,subpage_prot,swapoff,swapon,switch_endian,sys_debug_setcontext,sysfs,syslog,tuxcall,ulimit,umount,umount2,uselib,userfaultfd,ustat,vhangup,vm86,vm86old,vmspliceandvserver
```
For me: Fedora29 with qgis 2.18.20
qgis current: 3.6.X
but for example debian buster still has qgis 2.18.28.
https://packages.debian.org/buster/qgis

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**